common/network-common/src/main/java/org/apache/spark/network/sasl/aes/AesCipher.java

```diff
@@ -0,0 +1,340 @@
+      Properties properties,
+      byte[] inKey,
+      byte[] outKey,
+      byte[] inIv,
+      byte[] outIv) throws IOException {
+    properties.setProperty(CryptoInputStream.STREAM_BUFFER_SIZE_KEY,
+      String.valueOf(STREAM_BUFFER_SIZE));
+    this.properties = properties;
+
+    inputStreamMap = new HashMap<>();
+    outputStreamMap= new HashMap<>();
+
+    inKeySpec = new SecretKeySpec(inKey, "AES");
+    inIvSpec = new IvParameterSpec(inIv);
+    outKeySpec = new SecretKeySpec(outKey, "AES");
+    outIvSpec = new IvParameterSpec(outIv);
+  }
+
+  public AesCipher(AesConfigMessage configMessage) throws IOException  {
+    this(new Properties(), configMessage.inKey, configMessage.outKey,
+      configMessage.inIv, configMessage.outIv);
+  }
+
+  /**
+   * Create AES crypto output stream
+   * @param ch The underlying channel to write out.
+   * @return Return output crypto stream for encryption.
+   * @throws IOException
+   */
+  public CryptoOutputStream CreateOutputStream(WritableByteChannel ch) throws IOException {
+    if (!outputStreamMap.containsKey(ch)) {
+      outputStreamMap.put(ch, new CryptoOutputStream(TRANSFORM, properties, ch, outKeySpec, outIvSpec));
+    }
+
+    return outputStreamMap.get(ch);
+  }
+
+  /**
+   * Create AES crypto input stream
+   * @param ch The underlying channel used to read data.
+   * @return Return input crypto stream for decryption.
+   * @throws IOException
+   */
+  public CryptoInputStream CreateInputStream(ReadableByteChannel ch) throws IOException {
+    if (!inputStreamMap.containsKey(ch)) {
+      inputStreamMap.put(ch, new CryptoInputStream(TRANSFORM, properties, ch, inKeySpec, inIvSpec));
+    }
+
+    return inputStreamMap.get(ch);
+  }
+
+  /**
+   * Add handlers to channel
+   * @param ch the channel for adding handlers
+   * @throws IOException
+   */
+  public void addToChannel(Channel ch) throws IOException {
+    ch.pipeline()
+      .addFirst(ENCRYPTION_HANDLER_NAME, new AesEncryptHandler(this))
+      .addFirst(DECRYPTION_HANDLER_NAME, new AesDecryptHandler(this));
+  }
+
+  /**
+   * Generate a request config message which send to remote peer.
+   * @param conf is the local transport configuration.
+   * @return Config message for sending.
+   */
+  public static AesConfigMessage requestConfigMessage(TransportConf conf) {
+    int keySize = conf.saslEncryptionAesCipherKeySizeBits();
+    if (keySize % 8 != 0) {
+      throw new IllegalArgumentException("The AES cipher key size in bits should be a multiple " +
+        "of byte");
+    }
+    return new AesConfigMessage(keySize/8, null, null, null, null);
+  }
+
+  /**
+   * Generate the configuration message according to request config message.
+   * @param configMessage The request config message comes from remote.
+   * @return Configuration message for sending.
+   */
+  public static AesConfigMessage responseConfigMessage(AesConfigMessage configMessage){
+
+    Properties properties = new Properties();
```

Comment:
Why are these empty properties being created? Why isn't the transport config being used?